lib/controllers/DeviceController.ts

```diff
@@ -85,6 +85,29 @@ export class DeviceController extends CRUDController {
     };
   }
 
+  async update (request: KuzzleRequest) {
+    const deviceId = request.getId();
+    const body = request.getBody();
+    const devices = await this.mGetDevice([{ deviceId }]);
+
+    const response = await global.app.trigger(
+      'device-manager:device:update:before', {
+      device: devices[0],
+      updates: body,
+    });
+
+    request.input.body = response.updates;
+    const result = await super.update(request);
+
+    await global.app.trigger(
+      'device-manager:device:update:after', {
+      device: devices[0],
+      updates: result._source,
+    });
```

Comment:
Nitpicking: usually the construction for method having an object as last argument is the following
```suggestion
    await global.app.trigger('device-manager:device:update:after', {
      device: devices[0],
      updates: result._source,
    });
```